packages/terra-dropdown-button/src/_DropdownList.jsx

```diff
@@ -63,12 +64,20 @@ class DropdownList extends React.Component {
       event.preventDefault();
     } else if (keyCode === KeyCode.KEY_DOWN) {
       if (!this.pressed) {
-        this.changeFocusState(Util.findNext(this, this.state.focused));
+        if (index === Util.getChildArray(this).length - 1) {
+          this.changeFocusState(0);
+        } else {
+          this.changeFocusState(Util.findNext(this, this.state.focused));
+        }
       }
       event.preventDefault();
     } else if (keyCode === KeyCode.KEY_UP) {
       if (!this.pressed) {
-        this.changeFocusState(Util.findPrevious(this, this.state.focused));
+        if (index === 0) {
+          this.changeFocusState(Util.getChildArray(this).length - 1);
+        } else {
+          this.changeFocusState(Util.findPrevious(this, this.state.focused));
+        }
```

Comment:
Can this also be tested through Jest unit tests?

Reply:
This scenario has already been tested with WDIO since it is more related to the function like how the focus is set on the keydown. WDIO was more suitable for this as it provides snapshot evidence of the item being focused rather than validating on the Jest.

Reply:
Can we add some tests for other/edge cases (can be Jest or WDIO) such as:

- index != 0
- There are only 1 or 2 items in the dropdown
- The list is empty

Reply:
https://user-images.githubusercontent.com/126655256/233158147-273b4949-d654-41aa-953c-ae65a9eea496.mp4

Reply:
https://user-images.githubusercontent.com/126655256/233160783-4f6c530f-f490-402c-b231-b8e377f9db87.mp4

Reply:
Thanks for adding the screen recording! Can we also add a WDIO test for the 1 dropdown item case?

Reply:
Added testcase for 1 dropdown item